lib/commands/publish.js

```diff
@@ -1,6 +1,179 @@
 'use strict'
 
+const assert = require('assert').strict
+const pino = require('pino')
+const { validate } = require('../validation')
+const draft = require('./draft')
+const Npm = require('../npm')
+const GitDirectory = require('../git-directory')
+const GitHub = require('../github')
+
+// TODO this could be optimized with a common schema using $ref..
+const ARGS_SCHEMA = {
+  type: 'object',
+  required: ['path', 'verbose', 'major', 'remote', 'branch', 'semver', 'ghToken'],
+  properties: {
+    major: { type: 'boolean' },
+    help: { type: 'boolean' },
+    dryRun: { type: 'boolean' },
+    ghReleaseDraft: { type: 'boolean' },
+    ghReleasePrerelease: { type: 'boolean' },
+    path: { type: 'string' },
+    tag: { type: 'string' },
+    remote: { type: 'string' },
+    branch: { type: 'string' },
+    ghToken: {
+      type: 'string',
+      minLength: 40
+    },
+    verbose: {
+      type: 'string',
+      enum: ['debug', 'info', 'warn', 'error']
+    },
+    npmAccess: {
+      type: 'string',
+      enum: ['public', 'restricted']
+    },
+    npmDistTag: {
+      type: 'string',
+      pattern: '^[^v0-9][\\w]{1,12}'
+    },
+    semver: {
+      type: 'string',
+      enum: ['major', 'premajor', 'minor', 'preminor', 'patch', 'prepatch', 'prerelease']
+    }
+  }
+}
+
+const CLEAN_REPO = {
+  not_added: [],
+  conflicted: [],
+  created: [],
+  deleted: [],
+  modified: [],
+  renamed: [],
+  files: [],
+  staged: [],
+  ahead: 0,
+  behind: 0,
+  tracking: null
+}
+
+// TODO refactor to make optionals step (publish / push / release)
 module.exports = async function (args) {
-  // TODO
-  // [NOTE for PUBLISH command] if major => block it (unless there is a boolean option)
+  validate(args, ARGS_SCHEMA)
+
+  const logger = pino({ level: args.verbose, prettyPrint: true, base: null })
```

Comment:
I would move this in a utility, and move it to log to stderr instead

Reply:
Do you mean to log (all levels) to stderr instead of stdout?
Maybe releated to #13 ?